eslint.config.mjs

```diff
@@ -51,7 +68,22 @@ export default [
       'array-callback-return': 'off',
       'new-cap': 'off',
       'no-return-assign': 'error',
-      'object-shorthand': 'error'
+      'object-shorthand': 'error',
+      'no-constant-condition': 'off'
```

Comment:
We're turning off a lot of TS lints in this change.

Reply:
Actually we're not, these are added via `recommended-type-checked` and I believe we didn't have them enabled before. 

```
$ git diff
diff --git a/eslint.config.mjs b/eslint.config.mjs
index 53e6a18..acb7a4b 100644
--- a/eslint.config.mjs
+++ b/eslint.config.mjs
@@ -52,6 +52,13 @@ export default [
           allowRuleToRunWithoutStrictNullChecksIKnowWhatIAmDoing: false
         }
       ],
+      '@typescript-eslint/no-explicit-any': 'error',
+      '@typescript-eslint/no-unsafe-assignment': 'error',
+      '@typescript-eslint/no-unsafe-return': 'error',
+      '@typescript-eslint/no-unsafe-call': 'error',
+      '@typescript-eslint/no-unsafe-member-access': 'error',
+      '@typescript-eslint/no-unused-vars': ['error', { "argsIgnorePattern": "^_" }],
+      '@typescript-eslint/require-await': 'error',
       'array-callback-return': 'off',
       'new-cap': 'off',
       'no-return-assign': 'error',
```

```
$ npm run lint

✖ 193 problems (193 errors, 0 warnings)
  3 errors and 0 warnings potentially fixable with the `--fix` option.
```

Reply:
Two of these are easy to enable. The others give hundreds of errors.

- [opensearch-api-specification#318](https://github.com/opensearch-project/opensearch-api-specification/pull/318): Turn on @typescript-eslint/no-unused-var. 
- [opensearch-api-specification#319](https://github.com/opensearch-project/opensearch-api-specification/pull/319): Turn on @typescript-eslint/require-await.